node/cli/src/command.rs

```diff
@@ -0,0 +1,797 @@
+									params.accounts,
+								)),
+								config.network,
+							)
+						}
+						#[cfg(not(feature = "with-moonbeam-runtime"))]
+						return Err(MISSING_MOONBEAM.into());
+					} else if config.chain_spec.is_moonriver() {
+						#[cfg(feature = "with-moonriver-runtime")]
+						{
+							params.base.run(
+								Box::new(chain_spec::moonriver::development_chain_spec(
+									params.mnemonic.clone(),
+									params.accounts,
+								)),
+								config.network,
+							)
+						}
+						#[cfg(not(feature = "with-moonriver-runtime"))]
+						return Err(MISSING_MOONRIVER.into());
+					} else {
+						#[cfg(feature = "with-moonbase-runtime")]
+						{
+							params.base.run(
+								Box::new(chain_spec::moonbase::development_chain_spec(
+									params.mnemonic.clone(),
+									params.accounts,
+								)),
+								config.network,
+							)
+						}
+						#[cfg(not(feature = "with-moonbase-runtime"))]
+						return Err(MISSING_MOONBASE.into());
+					}
+				} else {
+					params.base.run(config.chain_spec, config.network)
+				}
+			})
+		}
+		Some(Subcommand::CheckBlock(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			runner.async_run(|mut config| {
+				let (client, _, import_queue, task_manager) =
+					service::new_chain_ops(&mut config, author_id)?;
+				Ok((cmd.run(client, import_queue), task_manager))
+			})
+		}
+		Some(Subcommand::ExportBlocks(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			runner.async_run(|mut config| {
+				let (client, _, _, task_manager) = service::new_chain_ops(&mut config, author_id)?;
+				Ok((cmd.run(client, config.database), task_manager))
+			})
+		}
+		Some(Subcommand::ExportState(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			runner.async_run(|mut config| {
+				let (client, _, _, task_manager) = service::new_chain_ops(&mut config, author_id)?;
+				Ok((cmd.run(client, config.chain_spec), task_manager))
+			})
+		}
+		Some(Subcommand::ImportBlocks(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			runner.async_run(|mut config| {
+				let (client, _, import_queue, task_manager) =
+					service::new_chain_ops(&mut config, author_id)?;
+				Ok((cmd.run(client, import_queue), task_manager))
+			})
+		}
+		Some(Subcommand::PurgeChain(cmd)) => {
+			// TODO-multiple-runtimes
+			let runner = cli.create_runner(cmd)?;
+
+			runner.sync_run(|config| {
+				// Although the cumulus_client_cli::PurgeCommand will extract the relay chain id,
+				// we need to extract it here to determine whether we are running the dev service.
+				let extension = {
+					#[cfg(feature = "with-moonbase-runtime")]
+					{
+						chain_spec::moonbase::Extensions::try_get(&*config.chain_spec)
+					}
+					#[cfg(feature = "with-moonbeam-runtime")]
+					{
+						chain_spec::moonbeam::Extensions::try_get(&*config.chain_spec)
+					}
+					#[cfg(feature = "with-moonriver-runtime")]
+					{
+						chain_spec::moonriver::Extensions::try_get(&*config.chain_spec)
+					}
+				};
```

Comment:
I wonder if this really needs to be conditional. I think that any one of them would work because extension is part of the spec that is outside of the runtime. Might be worth trying?